modules/relayers/src/payment_adapter.rs

```diff
@@ -20,65 +20,61 @@
 use crate::{Config, RelayerRewards};
 
 use bp_messages::source_chain::{MessageDeliveryAndDispatchPayment, RelayersRewards};
-use frame_support::{sp_runtime::SaturatedConversion, traits::Get};
-use sp_arithmetic::traits::{Saturating, Zero};
+use frame_support::sp_runtime::SaturatedConversion;
+use sp_arithmetic::traits::{Saturating, UniqueSaturatedFrom, Zero};
 use sp_std::{collections::vec_deque::VecDeque, marker::PhantomData, ops::RangeInclusive};
 
 /// Adapter that allows relayers pallet to be used as a delivery+dispatch payment mechanism
 /// for the messages pallet.
-pub struct MessageDeliveryAndDispatchPaymentAdapter<T, MessagesInstance, GetConfirmationFee>(
-	PhantomData<(T, MessagesInstance, GetConfirmationFee)>,
+pub struct MessageDeliveryAndDispatchPaymentAdapter<T, MessagesInstance>(
+	PhantomData<(T, MessagesInstance)>,
 );
 
-impl<T, MessagesInstance, GetConfirmationFee>
+impl<T, MessagesInstance>
 	MessageDeliveryAndDispatchPayment<T::RuntimeOrigin, T::AccountId, T::Reward>
-	for MessageDeliveryAndDispatchPaymentAdapter<T, MessagesInstance, GetConfirmationFee>
+	for MessageDeliveryAndDispatchPaymentAdapter<T, MessagesInstance>
 where
 	T: Config + pallet_bridge_messages::Config<MessagesInstance, OutboundMessageFee = T::Reward>,
 	MessagesInstance: 'static,
-	GetConfirmationFee: Get<T::Reward>,
 {
 	type Error = &'static str;
 
-	fn pay_delivery_and_dispatch_fee(
-		_submitter: &T::RuntimeOrigin,
-		_fee: &T::Reward,
-	) -> Result<(), Self::Error> {
-		// nothing shall happen here, because XCM deals with fee payment (planned to be burnt?
-		// or transferred to the treasury?)
-		Ok(())
-	}
-
 	fn pay_relayers_rewards(
-		lane_id: bp_messages::LaneId,
+		_lane_id: bp_messages::LaneId,
 		messages_relayers: VecDeque<bp_messages::UnrewardedRelayer<T::AccountId>>,
 		confirmation_relayer: &T::AccountId,
 		received_range: &RangeInclusive<bp_messages::MessageNonce>,
 	) {
 		let relayers_rewards = pallet_bridge_messages::calc_relayers_rewards::<T, MessagesInstance>(
-			lane_id,
 			messages_relayers,
 			received_range,
 		);
 
 		register_relayers_rewards::<T>(
 			confirmation_relayer,
 			relayers_rewards,
-			GetConfirmationFee::get(),
+			// TODO (https://github.com/paritytech/parity-bridges-common/issues/1318): this shall be fixed
```

Comment:
Shall be fixed either in this PR, or later

Reply:
Left it for later. New relayer rewards scheme will be introduced later && it deserves a separate review